### ISSUE TITLE
Users with specific settings management permissions can't access the editor page unless they also have ManageSettings permission

### ISSUE THREAD

**Issue by barthamark** [OrchardCMS/OrchardCore#5328]
What I'd like to achive is to give ManageEmailSettings permission to a user so they can update STMP settings, however, I don't want them to edit general settings (e.g. site name). When I set up a role with only ManageEmailSettings permission without ManageSettings permission the user can see the admin menu item but gets an "An error occurred while executing this request." message when opens it. 

Didn't try other site settings-related permissions but I assume we'd get the same result if that is a site settings entity extension.

**Comment by deanmarcussen** [comment]
So what it probably needs is a `ManageSiteSettings` permission to separate out the site permissions.

Then you can give `ManageSettings` permission as an overall permission, `ManageEmailSettings` for managing email but not `ManageSiteSettings`

Just requires a new Permission in the Settings project, and an `_authorize` line in `DefaultSiteSettingsDriver`

Be a good pr @barthamark

**Comment by sebastienros** [comment]
What I see here is that we could add another logic to "Imply" a permission. We can already have permissions implied by others. This way we could say that `ManageEmailSettings` also means `ManageSettings`, which would only be used to render a menu item.

**Comment by barthamark** [comment]
I don't want ManageEmailSettings to also mean ManageSettings. In other words I'd like to have users **with** ManageEmailSettings **without** ManageSettings (i.e. permission to edit SiteName etc.).

**Comment by jeffolmstead** [comment]
I have run into this also. I have a user I just want to have SMTP access. Even though I check them to have that permission in the role, they are not given it due to not having "Manage Settings" permissions (which I don't want to give them).
 
On a similar front, the admin menu shows they can click on "Security --> Settings --> Login" and "Security --> Settings --> Reset password" even though they don't have permissions for those pages. The pages result in an error telling them they don't have access to the resource if they click on them. Ideally though, the menu would reflect their actual permissions.

**Comment by mroskamp** [comment]
I've also run into this, basically the same scenario described in #7561:
* I add a custom site setting tied to a permission
* I add a role that's limited to only access the admin panel and manage my custom site setting
* The user is blocked from accessing the `SettingsController.IndexAsync` action unless they're also granted the `ManageSettings` permission
* Granting the `ManageSettings` permission also gives access to the "General" and "Debugging" settings, which is broader access than desired.

As @deanmarcussen [mentioned](https://github.com/OrchardCMS/OrchardCore/issues/5328#issuecomment-576211333), the crux of the issue is that [OrchardCore.Settings.AdminMenu](https://github.com/OrchardCMS/OrchardCore/blob/c72e72a50582895fa383e1d6bce3ed462d6756f3/src/OrchardCore.Modules/OrchardCore.Settings/AdminMenu.cs) ties those admin menu items to the generic `ManageGroupSettings` permission, rather than creating a more specific permission for authorizing the admin menu item and display driver, like most other settings do.

I've worked around this in my code by overriding/replacing the relevant display drivers and admin menu providers to check the custom permissions (`ManageGeneralSettings` and `ManageDebuggingSettings`).

What's the intended behavior here? Should users be able to manage individual settings groups without requiring explicit `ManageSettings` permissions, or is the current behavior by design and the General/Debugging settings simply missing more granular permissions?

If it's the latter, I'm happy to submit a PR to introduce dedicated permissions for the built-in General and Debugging settings and update the menu authorization logic accordingly.

**Comment by gvkries** [comment]
I think the original intent was:

- `ManageSettings` acts as a **global/master** permission for all settings.
- `ManageGroupSettings` is mainly intended to secure **custom settings** backed by content types. Internally, `CustomSettingsAuthorizationHandler` checks for permissions named `ManageCustomSettings_{0}` (with the group name added) when authorizing `ManageGroupSettings`.

The gap seems to be that settings exposed through `SiteDisplayDriver` aren’t consistently covered by that model, which is why scenarios like `ManageEmailSettings` still fail unless `ManageSettings` is also granted.

I agree with @sebastienros that the long-term fix is likely a broader permission-model change.  
Simply adding more granular settings permissions may not be enough, because those permissions still need to be honored directly by the relevant authorization checks (while also being implied by `ManageSettings`).

In short: we likely need module-specific settings permissions (e.g., `ManageEmailSettings`) to be first-class in the authorization flow for their corresponding settings editors, with `ManageSettings` remaining an optional super-permission.

**Comment by gvkries** [comment]
Here’s the idea I had in mind: #19661. But for now, it’s just an experiment.

**Comment by sebastienros** [comment]
Or mine https://github.com/OrchardCMS/OrchardCore/pull/19662
